src/js/angular/backup-and-restore/controllers.js

```diff
@@ -0,0 +1,65 @@
+const modules = [];
+
+angular
+    .module('graphdb.framework.monitoring.backupandrestore.controllers', modules)
+    .controller('BackupAndRestoreCtrl', BackupAndRestoreCtrl);
+
+BackupAndRestoreCtrl.$inject = ['$scope', '$interval', 'MonitoringRestService'];
+
+function BackupAndRestoreCtrl($scope, $interval, MonitoringRestService) {
+
+    $scope.loading = false;
+    $scope.initialized = false;
+
+    /**
+     * @type {BackupAndRestoreInfo[]}
+     */
+    $scope.backupAndRestoreInfos = undefined;
+    $scope.hasClusterOperation = false;
+    let timer = undefined;
+    const UPDATE_BACKUP_AND_RESTORE_INFO_DATA_TIME_INTERVAL = 2000;
+
+    // =========================
+    // Public functions
+    // =========================
+    $scope.hasToShowValue = (value) => {
+        if (typeof value === 'boolean') {
+            return true;
+        }
+        return !!value;
+    };
+
+    // =========================
+    // Private functions
+    // =========================
+    const loadBackupAndRestoreData = () => {
+        if ($scope.loading) {
+            return;
+        }
+        $scope.loading = true;
+        MonitoringRestService.monitorBackup()
+            .then((backupAndRestoreInfos) => {
+                $scope.backupAndRestoreInfos = backupAndRestoreInfos;
+                $scope.initialized = true;
+                $scope.hasClusterOperation = $scope.backupAndRestoreInfos.some((backupAndRestoreInfo) => backupAndRestoreInfo.nodePerformingClusterBackup);
+            })
+            .finally(() => $scope.loading = false);
+    };
+
+    const removeAllListeners = () => {
+        if (timer) {
+            $interval.cancel(timer);
+        }
+    };
+
+    // =========================
+    // Event handlers
+    // =========================
+    $scope.$on('$destroy', removeAllListeners);
+
+    // =========================
+    // Initializes controller.
+    // =========================
+    loadBackupAndRestoreData();
+    timer = $interval(() => loadBackupAndRestoreData(), UPDATE_BACKUP_AND_RESTORE_INFO_DATA_TIME_INTERVAL);
```

Comment:
If the monitor endpoint stalls for some reason, there will be multiple requests, which will resolve out of order and might cause some inconsistencies.

You can chain the next call to the previous. I think in the cluster or in the system monitor controller it is done like this, to prevent the issue.

Reply:
aa, you have a loading flag so this will not happen.